server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java

```diff
@@ -111,13 +111,15 @@ private void update(MessageId messageId, UpdateMessagePatch updateMessagePatch,
             }
         } catch (MailboxException e) {
             handleMessageUpdateException(messageId, builder, e);
+        } catch (IllegalStateException e) {
+            handleMessageUpdateException(messageId, builder, e);
         }
 
     }
 
     private Stream<MailboxException> updateFlags(MessageId messageId, UpdateMessagePatch updateMessagePatch, MailboxSession mailboxSession, MessageResult messageResult) {
         try {
-            if (!updateMessagePatch.isFlagsIdentity()) {
+            if (!updateMessagePatch.isFlagsIdentity(messageResult.getFlags())) {
```

Comment:
This change set smells not good to me, as being *identity* is a property of a transformation, not of its arguments (IE, we do not need argument to know if the transformation we intend to do is identity)